app/controllers/pet_applications_controller.rb

```diff
@@ -5,37 +5,45 @@ def index
   end
 
   def update
-    applicant = Applicant.find(params[:applicant])
+    pet_application = PetApplication.where(pet_id: params[:pet_id],applicant_id: params[:applicant])
+
     pet = Pet.find(params[:pet_id])
-    if pet.status == 'adoptible' # "adoptable"
-      params[:status] = "pending"
-      if params[:approve_all]
-        applicant.pets.each do |pet|
-          add_pet(pet, applicant)
-        end
-        return redirect_to "/pets"
-      end
-      add_pet(pet, applicant)
+    if PetApplication.where(pet_id: pet.id).pluck(:approval_status).include?(1)
```

Comment:
Nice change to .include?(1) to check for the approval_status. Seems more reliable than using the previous logic!